patches/server/0739-Add-config-setting-for-logging-player-ip-addresses.patch

```diff
@@ -0,0 +1,93 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Noah van der Aa <ndvdaa@gmail.com>
+Date: Fri, 6 Aug 2021 11:14:09 +0200
+Subject: [PATCH] Add config setting for logging player ip addresses.
+
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperConfig.java b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+index 95d8cf449bf0439fddaf319246fef51e1570b68e..1030b4c2bcdd17ec74619349a8f941ea3cc9262e 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperConfig.java
+@@ -506,6 +506,11 @@ public class PaperConfig {
+         logNamedEntityDeaths = getBoolean("settings.log-named-entity-deaths", logNamedEntityDeaths);
+     }
+ 
++    public static boolean logPlayerIpAddresses = true;
++    private static void playerIpAddresses() {
++        logPlayerIpAddresses = getBoolean("settings.log-player-ip-addresses", logPlayerIpAddresses);
++    }
++
+     public static int itemValidationDisplayNameLength = 8192;
+     public static int itemValidationLocNameLength = 8192;
+     public static int itemValidationLoreLineLength = 8192;
+diff --git a/src/main/java/net/minecraft/network/protocol/PacketUtils.java b/src/main/java/net/minecraft/network/protocol/PacketUtils.java
+index bcf53ec07b8eeec7a88fb67e6fb908362e6f51b0..335756e9b1f552ebe2873f49ee43cb4c91ea83db 100644
+--- a/src/main/java/net/minecraft/network/protocol/PacketUtils.java
++++ b/src/main/java/net/minecraft/network/protocol/PacketUtils.java
+@@ -38,10 +38,11 @@ public class PacketUtils {
+                     // Paper start
+                     catch (Exception e) {
+                         Connection networkmanager = listener.getConnection();
++                        String playerIP = com.destroystokyo.paper.PaperConfig.logPlayerIpAddresses ? String.valueOf(networkmanager.getRemoteAddress()) : ""; // Paper
+                         if (networkmanager.getPlayer() != null) {
+-                            LOGGER.error("Error whilst processing packet {} for {}[{}]", packet, networkmanager.getPlayer().getScoreboardName(), networkmanager.getRemoteAddress(), e);
++                            LOGGER.error("Error whilst processing packet {} for {}{}", packet, networkmanager.getPlayer().getScoreboardName(), playerIP.isEmpty() ? "" : "[" + playerIP + "]", e); // Paper
+                         } else {
+-                            LOGGER.error("Error whilst processing packet {} for connection from {}", packet, networkmanager.getRemoteAddress(), e);
++                            LOGGER.error("Error whilst processing packet {} for connection from {}", packet, playerIP, e); // Paper
```

Comment:
Same thing here about having the IP be an empty string. Would make the log entry look weird